hooks/_common.sh

```
@@ -606,3 +606,35 @@ function common::export_provided_env_vars {
+#   - 1 if version < 0.78.0
+#    Defaults to 0 if version cannot be determined
+#######################################################################
+# TODO: Drop after May 2027. Two years to upgrade is more than enough.
+function common::terragrunt_version_ge_0.78 {
+  local terragrunt_version
+
+  # Extract version number (e.g., "terragrunt version v0.80.4" -> "0.80")
+  terragrunt_version=$(terragrunt --version 2> /dev/null | grep -oE '[0-9]+\.[0-9]+')
+  # If we can't parse version, default to newer command
+  [[ ! $terragrunt_version ]] && return 0
+  fi
```

Shellcheck (0.10.0)

[error] Expected a '}'. If you have one, try a ; or \n in front of it.

(SC1056)